--- DyninstAPI-9.0.3/CMakeLists.txt.orig	2015-08-26 18:13:07.000000000 +0200
+++ DyninstAPI-9.0.3/CMakeLists.txt	2015-10-25 08:14:36.270127446 +0100
@@ -62,9 +62,7 @@
 if(NOT ${PLATFORM} MATCHES nt)
   add_subdirectory (elf)
   add_subdirectory (dwarf)
-if(SYMREADER MATCHES symLite)
   add_subdirectory (symlite)
-endif()
 # DynC is not compatible with Windows when
 # we use bison on linux to generate source
 # TODO: generate with bison configured for
--- DyninstAPI-9.0.3/symlite/CMakeLists.txt.orig	2015-08-26 18:13:07.000000000 +0200
+++ DyninstAPI-9.0.3/symlite/CMakeLists.txt	2015-10-25 14:59:53.432969271 +0100
@@ -18,6 +18,8 @@
 add_library (symLite_static STATIC ${SRC_LIST})
 FILE (GLOB headers "h/*.h")
 set_target_properties (symLite symLite_static PROPERTIES PUBLIC_HEADER "${headers}")
+set_target_properties (symLite PROPERTIES SOVERSION ${SOVERSION} VERSION ${LIBVERSION})
+set_target_properties (symLite_static PROPERTIES OUTPUT_NAME symLite)
 
 target_link_private_libraries (symLite common)
 target_link_private_libraries (symLite dynElf)
