--- dyninst-9.3.1/common/CMakeLists.txt.orig	2017-04-09 19:23:42.065823975 +0200
+++ dyninst-9.3.1/common/CMakeLists.txt	2017-04-09 19:28:45.709153841 +0200
@@ -112,6 +112,7 @@
 
 
 dyninst_library(common ${DEPS})
+set_target_properties(common PROPERTIES OUTPUT_NAME dyncommon)
 if(PLATFORM MATCHES nt OR PLATFORM MATCHES windows)
   target_link_private_libraries(common Psapi WS2_32 dbghelp)
 endif()
