--- dyninst-9.2.0/common/CMakeLists.txt.orig	2016-08-01 18:54:27.138427786 +0200
+++ dyninst-9.2.0/common/CMakeLists.txt	2016-08-01 18:56:45.575092871 +0200
@@ -112,6 +112,7 @@
 endif()
 
 dyninst_library(common ${DEPS})
+set_target_properties(common PROPERTIES OUTPUT_NAME dyncommon)
 if (UNIX)
   # Boost auto-links on Windows; don't double-link
   target_link_private_libraries(common ${Boost_LIBRARIES})
