--- DyninstAPI-8.1.2/common/make.module.tmpl.orig	2013-06-07 19:29:26.000000000 +0200
+++ DyninstAPI-8.1.2/common/make.module.tmpl	2013-07-12 16:01:20.253626485 +0200
@@ -29,9 +29,9 @@
 
 ifndef TARGET
 ifndef STATIC_COMPS
-TARGET		= libcommon.so
+TARGET		= libdyncommon.so
 else
-TARGET      = libcommon.a
+TARGET      = libdyncommon.a
 endif
 endif
 
--- DyninstAPI-8.1.2/instructionAPI/make.module.tmpl.orig	2013-06-07 19:29:35.000000000 +0200
+++ DyninstAPI-8.1.2/instructionAPI/make.module.tmpl	2013-07-12 17:17:31.730101306 +0200
@@ -16,7 +16,7 @@
 CFLAGS		     += $(USEFULWARNINGS) $(DEFINES)
 CXXFLAGS	     += $(USEFULWARNINGS) $(DEFINES)
 
-LDFLAGS += -lcommon
+LDFLAGS += -ldyncommon
 
 ifndef TARGET
 ifdef STATIC_COMPS
--- DyninstAPI-8.1.2/symtabAPI/make.module.tmpl.orig	2013-06-07 19:29:41.000000000 +0200
+++ DyninstAPI-8.1.2/symtabAPI/make.module.tmpl	2013-07-12 17:53:04.256678481 +0200
@@ -45,7 +45,7 @@
 endif
 endif
 
-LDFLAGS     += -L../../common/$(PLATFORM) -lcommon
+LDFLAGS     += -L../../common/$(PLATFORM) -ldyncommon
 
 ifdef LIBDWARF_PLATFORM
 LDFLAGS += -L../../dwarf/$(PLATFORM) -ldynDwarf
--- DyninstAPI-8.1.2/parseAPI/make.module.tmpl.orig	2013-06-07 19:29:36.000000000 +0200
+++ DyninstAPI-8.1.2/parseAPI/make.module.tmpl	2013-07-12 18:27:24.943258667 +0200
@@ -30,7 +30,7 @@
 CFLAGS		     += $(USEFULWARNINGS) -I$(TO_CORE)/../include
 CXXFLAGS	     += $(USEFULWARNINGS) -I$(TO_CORE)/../include
 
-LDFLAGS += -L../../common/$(PLATFORM) -lcommon -L../../symlite/$(PLATFORM) -lsymLite
+LDFLAGS += -L../../common/$(PLATFORM) -ldyncommon -L../../symlite/$(PLATFORM) -lsymLite
 
 ifneq (x$(WITHOUT_SYMTAB_API),xtrue)
 LDFLAGS += -L../../symtabAPI/$(PLATFORM) -lsymtabAPI
--- DyninstAPI-8.1.2/patchAPI/make.module.tmpl.orig	2013-06-07 19:29:38.000000000 +0200
+++ DyninstAPI-8.1.2/patchAPI/make.module.tmpl	2013-07-12 19:13:56.526474847 +0200
@@ -14,7 +14,7 @@
 CFLAGS		     += $(USEFULWARNINGS) -I$(TO_CORE)/../include
 CXXFLAGS	     += $(USEFULWARNINGS) -I$(TO_CORE)/../include
 
-LDFLAGS += -L../../common/$(PLATFORM) -L../../symtabAPI/$(PLATFORM) -L../../parseAPI/$(PLATFORM) -L../../instructionAPI/$(PLATFORM) -lcommon -lsymtabAPI -linstructionAPI -lparseAPI
+LDFLAGS += -L../../common/$(PLATFORM) -L../../symtabAPI/$(PLATFORM) -L../../parseAPI/$(PLATFORM) -L../../instructionAPI/$(PLATFORM) -ldyncommon -lsymtabAPI -linstructionAPI -lparseAPI
 
 LDFLAGS     += $(LIBDIR)
 ifndef USES_NATIVE_CC
--- DyninstAPI-8.1.2/Makefile.orig	2013-06-07 19:29:23.000000000 +0200
+++ DyninstAPI-8.1.2/Makefile	2013-07-12 20:16:12.919651379 +0200
@@ -166,14 +166,14 @@
 	@echo $(Everything)
 
 PKGCONFIG_CFLAGS = $(CXX_REQ_FLAGS) -I$(includedir)
-PKGCONFIG_LIBS = -L$(libdir) -ldyninstAPI -lstackwalk -lpcontrol -lpatchAPI -lparseAPI -linstructionAPI -lsymtabAPI -lsymLite -ldynDwarf -ldynElf -lcommon 
+PKGCONFIG_LIBS = -ldyninstAPI -lstackwalk -lpcontrol -lpatchAPI -lparseAPI -linstructionAPI -lsymtabAPI -lsymLite -ldynDwarf -ldynElf -ldyncommon 
 
 ifdef LIBDWARF_PLATFORM
 PKGCONFIG_CFLAGS += -I$(LIBDWARF_INC)
 ifeq (xtrue,x$(LIBDWARF_STATIC))
 PKGCONFIG_LIBS += -rdynamic -Wl,--whole-archive
 endif
-PKGCONFIG_LIBS += -L$(LIBDWARF_LIB) -ldwarf
+PKGCONFIG_LIBS += -ldwarf
 ifeq (xtrue,x$(LIBDWARF_STATIC))
 PKGCONFIG_LIBS += -Wl,--no-whole-archive
 endif
@@ -181,7 +181,7 @@
 
 ifdef LIBELF_PLATFORM
 PKGCONFIG_CFLAGS += -I$(LIBELF_INC)
-PKGCONFIG_LIBS += -L$(LIBELF_LIB) -lelf
+PKGCONFIG_LIBS += -lelf
 endif
 
 ifdef USE_LIBIBERTY
--- DyninstAPI-8.1.2/proccontrol/i386-unknown-linux2.4/Makefile.orig	2013-06-07 19:29:38.000000000 +0200
+++ DyninstAPI-8.1.2/proccontrol/i386-unknown-linux2.4/Makefile	2013-07-12 20:17:24.109648392 +0200
@@ -11,7 +11,7 @@
 
 TARGET = libpcontrol.so
 
-LDFLAGS     += $(LIBDIR) -lcommon
+LDFLAGS     += $(LIBDIR) -ldyncommon
 
 SRCS = 	../src/linux.C \
          ../src/unix.C \
--- DyninstAPI-8.1.2/proccontrol/x86_64-unknown-linux2.4/Makefile.orig	2013-06-07 19:29:39.000000000 +0200
+++ DyninstAPI-8.1.2/proccontrol/x86_64-unknown-linux2.4/Makefile	2013-07-13 12:37:17.537181065 +0200
@@ -11,7 +11,7 @@
 
 TARGET = libpcontrol.so
 
-LDFLAGS     += $(LIBDIR) -lcommon
+LDFLAGS     += $(LIBDIR) -ldyncommon
 
 SRCS =  ../src/linux.C \
         ../src/unix.C \
--- DyninstAPI-8.1.2/proccontrol/ppc32_linux/Makefile.orig	2013-06-07 19:29:38.000000000 +0200
+++ DyninstAPI-8.1.2/proccontrol/ppc32_linux/Makefile	2013-07-13 12:37:42.067180035 +0200
@@ -11,7 +11,7 @@
 
 TARGET = libpcontrol.so
 
-LDFLAGS     += $(LIBDIR) -lcommon
+LDFLAGS     += $(LIBDIR) -ldyncommon
 
 SRCS =  ../src/linux.C \
         ../src/unix.C \
--- DyninstAPI-8.1.2/proccontrol/ppc64_linux/Makefile.orig	2013-06-07 19:29:38.000000000 +0200
+++ DyninstAPI-8.1.2/proccontrol/ppc64_linux/Makefile	2013-07-13 12:37:59.953845953 +0200
@@ -11,7 +11,7 @@
 
 TARGET = libpcontrol.so
 
-LDFLAGS     += $(LIBDIR) -lcommon
+LDFLAGS     += $(LIBDIR) -ldyncommon
 
 SRCS = ../src/linux.C \
        ../src/unix.C \
--- DyninstAPI-8.1.2/stackwalk/i386-unknown-linux2.4/Makefile.orig	2013-06-07 19:29:40.000000000 +0200
+++ DyninstAPI-8.1.2/stackwalk/i386-unknown-linux2.4/Makefile	2013-07-12 20:35:22.019603157 +0200
@@ -17,7 +17,7 @@
 # Now make any necessary architecture specific changes to variables:
 include ../make.module.tmpl
 
-LDFLAGS     += $(LIBDIR) -lcommon -ldl
+LDFLAGS     += $(LIBDIR) -ldyncommon -ldl
 
 SRCS += 	../src/linux-swk.C \
 			../src/linuxbsd-x86-swk.C \
--- DyninstAPI-8.1.2/stackwalk/x86_64-unknown-linux2.4/Makefile.orig	2013-06-07 19:29:40.000000000 +0200
+++ DyninstAPI-8.1.2/stackwalk/x86_64-unknown-linux2.4/Makefile	2013-07-13 12:38:20.397178429 +0200
@@ -11,7 +11,7 @@
 
 include ../../make.config 
 
-LDFLAGS     += -lcommon
+LDFLAGS     += -ldyncommon
 
 SRCS = 	../src/linux-swk.C \
 	../src/linuxbsd-x86-swk.C \
--- DyninstAPI-8.1.2/stackwalk/ppc32_linux/Makefile.orig	2013-06-07 19:29:40.000000000 +0200
+++ DyninstAPI-8.1.2/stackwalk/ppc32_linux/Makefile	2013-07-13 12:38:44.507177415 +0200
@@ -14,7 +14,7 @@
 
 # Now make any necessary architecture specific changes to variables:
 
-LDFLAGS     += $(LIBDIR) -lcommon -ldl
+LDFLAGS     += $(LIBDIR) -ldyncommon -ldl
 
 SRCS = 	../src/linux-swk.C \
 	../src/linux-ppc-swk.C \
--- DyninstAPI-8.1.2/stackwalk/ppc64_linux/Makefile.orig	2013-06-07 19:29:40.000000000 +0200
+++ DyninstAPI-8.1.2/stackwalk/ppc64_linux/Makefile	2013-07-13 12:39:03.747176611 +0200
@@ -14,7 +14,7 @@
 
 # Now make any necessary architecture specific changes to variables:
 
-LDFLAGS     += $(LIBDIR) -lcommon -ldl
+LDFLAGS     += $(LIBDIR) -ldyncommon -ldl
 
 SRCS = 	../src/linux-swk.C \
 	../src/linux-ppc-swk.C \
