--- dyninst-12.2.0/cmake/Modules/FindTBB.cmake.orig	2022-08-23 05:33:29.156228610 +0200
+++ dyninst-12.2.0/cmake/Modules/FindTBB.cmake	2022-08-23 05:37:07.615045117 +0200
@@ -154,7 +154,8 @@ find_path(
 if(TBB_INCLUDE_DIRS)
     # Starting in 2020.1.1, tbb_stddef.h is replaced by version.h
     set(_version_files "${TBB_INCLUDE_DIRS}/tbb/tbb_stddef.h"
-                       "${TBB_INCLUDE_DIRS}/tbb/version.h")
+                       "${TBB_INCLUDE_DIRS}/tbb/version.h"
+                       "${TBB_INCLUDE_DIRS}/oneapi/tbb/version.h")
     foreach(f IN ITEMS ${_version_files})
         if(EXISTS ${f})
             set(_version_file ${f})
