diff -ur dyninst-10.1.0.orig/common/CMakeLists.txt dyninst-10.1.0/common/CMakeLists.txt
--- dyninst-10.1.0.orig/common/CMakeLists.txt	2019-05-16 20:40:05.000000000 +0200
+++ dyninst-10.1.0/common/CMakeLists.txt	2019-08-28 20:22:16.367000000 +0200
@@ -111,6 +111,7 @@
 
 
 dyninst_library(common ${DEPS})
+set_target_properties(common PROPERTIES OUTPUT_NAME dyncommon)
 if(TARGET TBB)
     add_dependencies(common TBB)
 endif()
