--- dyninst-12.2.0/cmake/Modules/FindTBB.cmake.orig	2022-08-23 05:33:29.156228610 +0200
+++ dyninst-12.2.0/cmake/Modules/FindTBB.cmake	2022-08-23 05:37:07.615045117 +0200
@@ -154,7 +154,8 @@ find_path(
 if(TBB_INCLUDE_DIRS)
     # Starting in 2020.1.1, tbb_stddef.h is replaced by version.h
     set(_version_files "${TBB_INCLUDE_DIRS}/tbb/tbb_stddef.h"
-                       "${TBB_INCLUDE_DIRS}/tbb/version.h")
+                       "${TBB_INCLUDE_DIRS}/tbb/version.h"
+                       "${TBB_INCLUDE_DIRS}/oneapi/tbb/version.h")
     foreach(f IN ITEMS ${_version_files})
         if(EXISTS ${f})
             set(_version_file ${f})
diff --color -ur dyninst-12.0.1.orig/common/h/concurrent.h dyninst-12.0.1/common/h/concurrent.h
--- dyninst-12.0.1.orig/common/h/concurrent.h	2021-11-23 22:48:27.000000000 +0100
+++ dyninst-12.0.1/common/h/concurrent.h	2021-12-20 00:43:49.261018867 +0100
@@ -54,10 +54,10 @@
 
 template<typename K, typename V>
 class dyn_c_hash_map : protected tbb::concurrent_hash_map<K, V,
-    tbb::tbb_hash_compare<K>, std::allocator<std::pair<K,V>>> {
+    tbb::tbb_hash_compare<K>, std::allocator<std::pair<const K,V>>> {
 
     typedef tbb::concurrent_hash_map<K, V,
-        tbb::tbb_hash_compare<K>, std::allocator<std::pair<K,V>>> base;
+        tbb::tbb_hash_compare<K>, std::allocator<std::pair<const K,V>>> base;
 public:
     using typename base::value_type;
     using typename base::mapped_type;
diff --color -ur dyninst-12.0.1.orig/symtabAPI/src/dwarfWalker.h dyninst-12.0.1/symtabAPI/src/dwarfWalker.h
--- dyninst-12.0.1.orig/symtabAPI/src/dwarfWalker.h	2021-11-23 22:48:27.000000000 +0100
+++ dyninst-12.0.1/symtabAPI/src/dwarfWalker.h	2021-12-20 00:46:32.186998418 +0100
@@ -36,6 +36,8 @@
 }
 
 namespace tbb {
+namespace detail {
+namespace d1 {
     using namespace Dyninst::SymtabAPI;
     template<>
     struct tbb_hash_compare<type_key> {
@@ -51,6 +53,8 @@
         }
     };
 }
+}
+}
 
 namespace Dyninst {
 namespace SymtabAPI {
