diff -ur dyninst-12.2.1.orig/instructionAPI/src/AMDGPU/cdna2/InstructionDecoder-amdgpu-cdna2.h dyninst-12.2.1/instructionAPI/src/AMDGPU/cdna2/InstructionDecoder-amdgpu-cdna2.h
--- dyninst-12.2.1.orig/instructionAPI/src/AMDGPU/cdna2/InstructionDecoder-amdgpu-cdna2.h	2022-11-21 20:46:38.000000000 +0100
+++ dyninst-12.2.1/instructionAPI/src/AMDGPU/cdna2/InstructionDecoder-amdgpu-cdna2.h	2023-02-19 21:33:54.774376195 +0100
@@ -33,6 +33,9 @@
 #include "Immediate.h"
 #include "dyn_regs.h"
 
+#include <array>
+#include <string>
+
 namespace Dyninst {
     namespace InstructionAPI {
 
diff -ur dyninst-12.2.1.orig/instructionAPI/src/AMDGPU/vega/InstructionDecoder-amdgpu-vega.h dyninst-12.2.1/instructionAPI/src/AMDGPU/vega/InstructionDecoder-amdgpu-vega.h
--- dyninst-12.2.1.orig/instructionAPI/src/AMDGPU/vega/InstructionDecoder-amdgpu-vega.h	2022-11-21 20:46:38.000000000 +0100
+++ dyninst-12.2.1/instructionAPI/src/AMDGPU/vega/InstructionDecoder-amdgpu-vega.h	2023-02-19 21:33:48.364566725 +0100
@@ -33,6 +33,9 @@
 #include "Immediate.h"
 #include "dyn_regs.h"
 
+#include <array>
+#include <string>
+
 namespace Dyninst {
     namespace InstructionAPI {
 
--- dyninst-12.3.0.orig/instructionAPI/src/InstructionDecoder-aarch64.h	2022-11-21 20:46:38.000000000 +0100
+++ dyninst-12.3.0/instructionAPI/src/InstructionDecoder-aarch64.h	2023-02-19 21:33:39.774822043 +0100
@@ -33,6 +33,8 @@
 #include "Immediate.h"
 #include "dyn_regs.h"
 
+#include <string>
+
 namespace Dyninst {
     namespace InstructionAPI {
 
